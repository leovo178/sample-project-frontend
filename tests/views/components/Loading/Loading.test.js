import React from 'react'

import { cleanup, render } from '@testing-library/react'

import { Loading } from '@views_components'

describe('<Loading />', () => {
	const mockProps = {
		open: true,
		msg: 'Loading...',
	}

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const { container, queryByText } = render(<Loading {...mockProps} />)

		expect(container).toMatchSnapshot()
		expect(queryByText(mockProps.msg)).toBeTruthy()
	})

	it('should render a <div/> when open is falsy', () => {
		const props = { ...mockProps, open: false }
		const { queryByText } = render(<Loading {...props} />)

		expect(queryByText(props.msg)).toBeFalsy()
	})
})
