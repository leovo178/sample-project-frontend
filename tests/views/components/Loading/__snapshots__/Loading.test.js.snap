// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    Loading...
  </div>
</div>
`;
